Algorithm.CSharp/BaseBitcoin.cs

```diff
@@ -0,0 +1,87 @@
+﻿using NodaTime;
+using QuantConnect.Brokerages;
+using QuantConnect.Data.Market;
+using QuantConnect.Securities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+
+    /// <summary>
+    /// Base class for Bitcoin algorithms. This base class provides standard behaviour for trading BTCUSD on Bitfinex
+    /// trading on Bitfinex
+    /// </summary>
+    public abstract class BaseBitcoin : QCAlgorithm
+    {
+
+        const string _bitcoinSymbol = "BTCUSD";
+
+        protected string BitcoinSymbol { get { return _bitcoinSymbol; } }
+
+        public BaseBitcoin()
+        {
+            Portfolio = new BitfinexSecurityPortfolioManager(Securities, Transactions);
+            SetBrokerageModel(BrokerageName.BitfinexBrokerage, AccountType.Margin);
```

Comment:
The margin call behavior is already plugable/extensible via the `IMarginModel.GenerateMarginCallOrder` - so users can define how margin calls happen. By default, we use a system modeled off of interactive broker's margin call behavior. If BTC has different margin rules then we should add a dedicated `IMarginModel` implementation, but defining a new portfolio type is not the place to do this as the type wasn't designed to be extended.